### PR TITLE
LOG-532. Vendor fluent-plugin-splunk-hec

### PR DESCRIPTION
**Pull request by jcantrill**
Ref: https://jira.coreos.com/browse/LOG-532

### CODE REVIEW

**Comment by richm** [comment]
hmm - well - I don't see a way to disable this signing stuff - https://guides.rubygems.org/security/#building-gems - I don't see a flag in `gem build` or `gem install` to completely disable signing.

Looks like `fluentd/vendored_gem_src/systemd-journal/systemd-journal.gemspec` does something similar - not sure why we don't have a problem there.

I think you are going to have to write a patch for ffi-compiler - similar to what we had to do for ffi https://github.com/openshift/origin-aggregated-logging/blob/master/fluentd/ffi.gemspec.patch.sh - basically, remove these two lines from the gemspec:
```
  s.cert_chain  = ['certs/stakach.pem']
  s.signing_key = File.expand_path("~/.ssh/gem-private_key.pem") if $0 =~ /gem\z/
```

**Comment by jcantrill** [comment]
@richm tests passed so merged commits

File: fluentd/Dockerfile [hunk 1]
```diff
@@ -36,14 +36,15 @@ ADD source.jemalloc /source.jemalloc
 RUN bash -c '. /source.jemalloc; echo jemalloc $JEMALLOC_VER >> /contents'
 
 ADD vendored_gem_src/ ${HOME}/vendored_gem_src/
+RUN mkdir -p ${HOME}/.ssh ; touch ${HOME}/.ssh/gem-private_key.pem
```
Comment (richm) [inline fluentd/Dockerfile]: you can remove this - not needed with the ffi-compiler patch

File: fluentd/Dockerfile.centos7 [hunk 2]
```diff
@@ -34,6 +34,7 @@ ADD source.jemalloc /source.jemalloc
 RUN bash -c '. /source.jemalloc; echo jemalloc $JEMALLOC_VER >> /contents'
 
 ADD vendored_gem_src/ ${HOME}/vendored_gem_src/
+RUN mkdir -p ${HOME}/.ssh ; touch ${HOME}/.ssh/gem-private_key.pem
```
Comment (richm) [inline fluentd/Dockerfile.centos7]: see above

**Comment by richm** [comment]
some sort of registry flake?
/retest

**Comment by richm** [comment]
multi-tenancy flake
/retest

**Comment by richm** [comment]
rsyslog build flake (should remove rsyslog . . .)
/retest

**Comment by richm** [comment]
hmm - another multi-tenancy test failure - might not be a flake?
/retest

**Comment by richm** [comment]
aws flake
/retest

**Pull request merged**